### PR TITLE
Adding the initial spec file version for rust-chariott

### PR DESCRIPTION
**Pull request by mrbrandao**
+ Initial version generated with rust2rpm
+ Added all the BuildRequires manually since chariott is not on crates yet

### CODE REVIEW

**Comment by odra** [comment]
Could you please `images/Containerfile` to `images/chariott/Containerfile` and `rpms/rust-chariott.spec` to `rpms/rust-chariott/rust-chariott.spec`?

File: images/Containerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+COPY --from=builder /sdv/target/release/chariott /usr/local/bin/chariott
+COPY --from=builder /sdv/target/release/chariott.d /etc/
+COPY --from=builder /sdv/target/debug/kv-app /usr/local/bin/kv-app
+COPY --from=builder /sdv/target/debug/kv-app.d /etc/
+RUN chown -R chariott:chariott /etc/chariott.d /etc/kv-app.d
+
+# Use the unprivileged chariott user during execution.
+USER chariott:chariott
+
+CMD ["chariott"]
```
Comment (odra) [inline images/Containerfile:37]: `ENTRYPOINT` makes more sense

File: images/Containerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+# Copy our build
+COPY --from=builder /sdv/target/release/chariott /usr/local/bin/chariott
+COPY --from=builder /sdv/target/release/chariott.d /etc/
+COPY --from=builder /sdv/target/debug/kv-app /usr/local/bin/kv-app
+COPY --from=builder /sdv/target/debug/kv-app.d /etc/
+RUN chown -R chariott:chariott /etc/chariott.d /etc/kv-app.d
+
+# Use the unprivileged chariott user during execution.
+USER chariott:chariott
```
Comment (odra) [inline images/Containerfile:35]: We don't need a chariott user/group for this base image

File: images/Containerfile [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Containerfile for building Eclipse Chariott runtime container
+FROM registry.fedoraproject.org/fedora:38 as builder
+ENV VERSION="56fac8c49eb19bc96dfab1ca10a6bd9a5a1985e4"
+RUN dnf -y install rust cargo unzip cmake protobuf-devel
+ADD https://github.com/eclipse-chariott/chariott/archive/${VERSION}.zip /tmp/chariott.zip
+RUN unzip /tmp/chariott.zip -d /tmp;mv /tmp/chariott-${VERSION} /sdv
+WORKDIR /sdv
+RUN cargo build --release 
+RUN cargo build -p kv-app
```
Comment (odra) [inline images/Containerfile:9]: Let's remove the kv-app build from this image and create another containerfile/image for it in a different PR.

File: images/Containerfile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+# Chariott Final Fedora Image
+FROM registry.fedoraproject.org/fedora:38
+
+# Chariott user id
+ARG CHARIOTT_UID=10001
```
Comment (odra) [inline images/Containerfile:15]: No need for uid creation as well

File: images/Containerfile (hunk 1 above)
Comment (mrbrandao) [reply images/Containerfile]: done. 
Take a look and let me know.

File: images/Containerfile (hunk 3 above)
Comment (mrbrandao) [reply images/Containerfile]: Fixed. I've removed the kv-app will create a new PR with a separated image

File: images/Containerfile (hunk 4 above)
Comment (mrbrandao) [reply images/Containerfile]: Fixed. Removed the UID and all user creation references as well

File: images/chariott/Containerfile [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# Containerfile for building Eclipse Chariott runtime container
+FROM registry.fedoraproject.org/fedora:38 as builder
+ENV VERSION="56fac8c49eb19bc96dfab1ca10a6bd9a5a1985e4"
```
Comment (odra) [inline images/chariott/Containerfile:3]: Can you change to an `ARG`? This way the  version can be parametrized from the cli: `podman build --build-arg VERSION=$something ...`

Comment (mrbrandao) [reply images/chariott/Containerfile]: Good catch. Added this support in the latest commit

**Pull request merged**